conda.recipe/meta.yaml

```diff
@@ -20,7 +20,7 @@ extra:
 requirements:
   build:
     - jinja2
-    - nodejs >=10.13
+    - nodejs >=10.13,<13
```

Comment:
What's the need for upper bound?

Reply:
on windows, at least sometimes, the `test_ext` fails because the `npm install` fails:

> 
> 2019-11-20T18:36:20.8687275Z ================================== FAILURES ===================================
> 2019-11-20T18:36:20.8782807Z ______________________________ test_ext_commands ______________________________
> 2019-11-20T18:36:20.8783473Z 
> 2019-11-20T18:36:20.8783972Z tmpdir = local('C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\test_ext_commands0')
> 2019-11-20T18:36:20.8784929Z 
> 2019-11-20T18:36:20.8785541Z     def test_ext_commands(tmpdir):
> 2019-11-20T18:36:20.8785820Z         tmp = str(tmpdir.mkdir("bk_ext_01"))
> 2019-11-20T18:36:20.8786043Z     
> 2019-11-20T18:36:20.8786586Z         assert _names(tmp) == []
> 2019-11-20T18:36:20.8786870Z     
> 2019-11-20T18:36:20.8787127Z         assert ext.init(tmp, bokehjs_version="1.3.4") is True
> 2019-11-20T18:36:20.8787375Z         assert _names(tmp) == [
> 2019-11-20T18:36:20.8787616Z             "bokeh.ext.json",
> 2019-11-20T18:36:20.8787851Z             "index.ts",
> 2019-11-20T18:36:20.8788104Z             "package.json",
> 2019-11-20T18:36:20.8788329Z             "tsconfig.json",
> 2019-11-20T18:36:20.8788548Z         ]
> 2019-11-20T18:36:20.8788756Z     
> 2019-11-20T18:36:20.8789039Z >       assert ext.build(tmp) is True
> 2019-11-20T18:36:20.8789342Z E       AssertionError: assert False is True
> 2019-11-20T18:36:20.8789890Z E        +  where False = <function build at 0x0000023175627730>('C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\test_ext_commands0\\bk_ext_01')
> 2019-11-20T18:36:20.8790770Z E        +    where <function build at 0x0000023175627730> = ext.build
> 2019-11-20T18:36:20.8791038Z 
> 2019-11-20T18:36:20.8791340Z tests\unit\bokeh\test_ext.py:48: AssertionError
> 2019-11-20T18:36:20.8791714Z ---------------------------- Captured stdout call -----------------------------
> 2019-11-20T18:36:20.8792157Z Working directory: C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\test_ext_commands0\bk_ext_01
> 2019-11-20T18:36:20.8792607Z Wrote C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\test_ext_commands0\bk_ext_01\bokeh.ext.json
> 2019-11-20T18:36:20.8793070Z Wrote C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\test_ext_commands0\bk_ext_01\package.json
> 2019-11-20T18:36:20.8793509Z Wrote C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\test_ext_commands0\bk_ext_01\tsconfig.json
> 2019-11-20T18:36:20.8793850Z Created empty index.ts. This is the entry point of your extension.
> 2019-11-20T18:36:20.8794651Z You can build your extension with bokeh build C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\test_ext_commands0\bk_ext_01
> 2019-11-20T18:36:20.8795467Z All done.
> 2019-11-20T18:36:20.8796733Z Working directory: C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\test_ext_commands0\bk_ext_01
> 2019-11-20T18:36:20.8797618Z Using different version of bokeh, rebuilding from scratch.
> 2019-11-20T18:36:20.8798700Z Running npm install.
> 2019-11-20T18:36:20.8799574Z npm install failed with exit code 1.
> 2019-11-20T18:36:20.8801277Z ---------------------------- Captured stderr call -----------------------------
> 2019-11-20T18:36:20.8801700Z npm WARN npm npm does not support Node.js v13.0.0
> 2019-11-20T18:36:20.8802069Z npm WARN npm You should probably upgrade to a newer version of node as we
> 2019-11-20T18:36:20.8802428Z npm WARN npm can't make any promises that npm will work with this version.
> 2019-11-20T18:36:20.8802810Z npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11, 12.
> 2019-11-20T18:36:20.8803202Z npm WARN npm You can find the latest version at https://nodejs.org/
> 2019-11-20T18:36:20.8803505Z npm ERR! Error while executing:
> 2019-11-20T18:36:20.8803923Z npm ERR! C:\Program Files\Git\mingw64\bin\git.EXE ls-remote -h -t https://github.com/bokeh/SlickGrid.git
> 2019-11-20T18:36:20.8804185Z npm ERR! 
> 2019-11-20T18:36:20.8804590Z npm ERR! fatal: remote error: access denied or repository not exported: /6/nw/60/d2/f6/440826/94910591.git
> 2019-11-20T18:36:20.8805030Z npm ERR! 
> 2019-11-20T18:36:20.8805308Z npm ERR! exited with error code: 128
> 2019-11-20T18:36:20.8805529Z 
> 2019-11-20T18:36:20.8805943Z npm ERR! A complete log of this run can be found in:
> 2019-11-20T18:36:20.8806318Z npm ERR!     C:\npm\cache\_logs\2019-11-20T18_34_32_658Z-debug.log
>

Reply:
It looked like that was complaining about the nodes version but it seems that's not the ultimate cause. I think it cane reverted.

Reply:
Only even versions of nodejs are relevant, so that fine for now. Though it would be great to add a commit message briefly explaining the reason or pointing to a discussion. Otherwise I would found myself at some point wondering why it was done, similarly to trying to figure out right now why selenium is pinned to version 3.8 (btw. do you remember why?).

Reply:
The actual error above there is some git issue on closer inspection, and I see a windows build with 0.13 did in fact pass at a later time, so I think the best thing is just to remove it. But agree a note would have been useful, thank you for suggesting. I don't offhand recall anything about selenium 3.8 tho a go-to assumption is usually some package availability problem. It's probably worth seeing if it can be lifted.